### ISSUE TITLE
Request connection params with callback from Android Oreo or higher

### ISSUE THREAD

**Issue by zetokz** [Polidea/RxAndroidBle#508]
Android Oreo or higher allows receiving gatt callback about connection params update. It would be good to rely on this callback when request connection parameters instead of just waiting for a timeout to finish subscription.

**Comment by dariuszseweryn** [comment]
I am looking into this request right now. I see a potential issue.

Right after connecting the OS changes connection parameters in order to speed up service discovery process. This behaviour can interfere with the proposed change. Any ideas?

**Comment by dariuszseweryn** [comment]
I have created a PR that adds possibility of using the new callback and creating a custom request connection priority operation. I have not wired it up with the current API due to no possibility of distinguishing if a callback is related to a change in parameters due to the user's request, the system changing it or the peripheral.

**Comment by dariuszseweryn** [comment]
Access to the updated parameters is available in `1.10.0` which should be available shortly. Unfortunately due to this not being a public API and an uncertain behaviour of the OS regarding params updates it is up to the user to implement it. E.g.
```java
@RequiresApi(26)
private Observable<ConnectionParameters> requestConnectionPriority(RxBleConnection rxBleConnection, int priority) {
    return Observable.merge(
            rxBleConnection.requestConnectionPriority(BluetoothGatt.CONNECTION_PRIORITY_HIGH, 1, TimeUnit.MILLISECONDS).toObservable(),
            rxBleConnection.observeConnectionParametersUpdates().take(1)
    );
}
```
